### PR TITLE
Make aten.copy preserve strides (hf_Longformer)

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89464

Fixes https://github.com/pytorch/pytorch/issues/93428

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D41460986](https://our.internmc.facebook.com/intern/diff/D41460986)

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Review by bdhirsh — APPROVED** [review]
There's a chance this might mobile, if they have the `copy_` -> `view_copy` decomp baked in anywhere.

**Comment by ezyang** [comment]
thx, I'll import

**Comment by ezyang** [comment]
@ezyang has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D41460986).

**Comment by ezyang** [comment]
@ezyang has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D41460986).

File: aten/src/ATen/native/Copy.cpp [hunk 1]
```diff
@@ -278,27 +278,39 @@ static Tensor & copy_impl(Tensor & self, const Tensor & src, bool non_blocking)
...
+  // with internal overlap.
+  //if (at::has_internal_overlap(self) == at::MemOverlap::Yes) {
+  //  return self.clone();
+  //}
+  auto dtype_size = self.dtype().itemsize();
+  auto nbytes = self.storage().sym_nbytes();
+  TORCH_INTERNAL_ASSERT(nbytes % dtype_size == 0);
+  auto numel = nbytes / dtype_size;
+  auto self_full_size = self.as_strided_symint({numel}, {1}, 0);
+  auto clone = self_full_size.clone();
```
Comment (ngimel) [inline aten/src/ATen/native/Copy.cpp:302]: are we ok with this potentially using much more memory than it should?

Comment (ezyang) [reply aten/src/ATen/native/Copy.cpp:302]: It won't matter in inductor because inductor will decompose throwing out the storage size.

**Comment by ezyang** [comment]
@ezyang has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D41460986).

**Comment by ezyang** [comment]
@ezyang has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D41460986).

**Pull request closed without merging**